pkg/component/kubernetes/apiserver/authentication.go

```diff
@@ -44,90 +45,113 @@ func (k *kubeAPIServer) reconcileConfigMapAuthenticationConfig(ctx context.Conte
 		return errors.New("oidc configuration is incompatible with structured authentication")
 	}
 
-	if value, ok := k.values.FeatureGates["StructuredAuthenticationConfiguration"]; (ok && !value) ||
-		(k.values.AuthenticationConfiguration == nil && k.values.OIDC == nil) ||
+	var (
+		structAuthGateEnabled, structAuthGateSet  = k.values.FeatureGates["StructuredAuthenticationConfiguration"]
+		anonymousGateEnabled, anonymousGateSet    = k.values.FeatureGates["AnonymousAuthConfigurableEndpoints"]
+		shouldConfigureAnonymousViaStructuredAuth = (!anonymousGateSet || (anonymousGateSet && anonymousGateEnabled)) && versionutils.ConstraintK8sGreaterEqual132.Check(k.values.Version)
```

Comment:
```suggestion
		shouldConfigureAnonymousViaStructuredAuth = (!anonymousGateSet || anonymousGateEnabled) && versionutils.ConstraintK8sGreaterEqual132.Check(k.values.Version)
```

Nit, to increase readability.